desafio-02/Mekylei-Belchior/python/desafio2.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+
+"""
+Desafio 02
+"""
+
+numeros = (n for n in range(1, 10001))
+primos = []
+
+for numero in numeros:
+    if numero == 2:
+        primos.append(numero)
+
+    if numero > 2:
+        divisores = (n for n in range(1, numero + 1))
```

Comment:
É melhor que a sua range seja (2..numero - 1) em python, range(2, numero). Voce já sabe que todo numero é divisível por 1 e por ele mesmo. Outra coisa, por que o list comprehension aqui? É melhor fazer direto:

`divisores = range(2, numero)`

Ou melhor ainda, elimine divisores completamente e no loop abaixo faça:

`for divisor in range(2, numero)`

Reply:
Neste caso, havia utilizado um gerador. Mas, optei por alterar conforme sugerido ao qual torna mais legível e eficiente.